### PR TITLE
[ARM] Fix #19468: `pip` installs `azure-cli` 2.0.73 because of the dependency on deprecated `jsmin`

### LINKED ISSUE #19468: `pip` installs `azure-cli` 2.0.73 because of the dependency on deprecated `jsmin`
I am trying to install the latest az cli, but instead I am getting an older version due to a jsmin setup error:

```
python -m pip install --upgrade wheel setuptools
python -m pip install --upgrade azure-cli azure-devops azure-keyvault-secrets azure-identity
python -m pip install --upgrade testresources
# python -m pip install --upgrade cryptography
python -m pip install --upgrade requests

2021-09-07T11:26:04.9965939Z   Downloading https://pkgs.dev.azure.com/mihaimyh/49255723-5232-4e9f-9501-068bf5e381a9/_packaging/e1661b2f-7e5d-484a-aab6-54c2a1cb5b0e/pypi/download/jsmin/2.2.2/jsmin-2.2.2.tar.gz (12 kB)
2021-09-07T11:26:05.2418035Z     ERROR: Command errored out with exit status 1:
2021-09-07T11:26:05.2421084Z      command: /usr/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-a863hy6i/jsmin_710895a4c3e44d21a48f56af2b8b8174/setup.py'"'"'; __file__='"'"'/tmp/pip-install-a863hy6i/jsmin_710895a4c3e44d21a48f56af2b8b8174/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-etdryiat
2021-09-07T11:26:05.2423588Z          cwd: /tmp/pip-install-a863hy6i/jsmin_710895a4c3e44d21a48f56af2b8b8174/
2021-09-07T11:26:05.2424522Z     Complete output (1 lines):
2021-09-07T11:26:05.2424978Z     error in jsmin setup command: use_2to3 is invalid.
2021-09-07T11:26:05.2425795Z     ----------------------------------------
2021-09-07T11:26:05.2429984Z WARNING: Discarding https://pkgs.dev.azure.com/mihaimyh/49255723-5232-4e9f-9501-068bf5e381a9/_packaging/e1661b2f-7e5d-484a-aab6-54c2a1cb5b0e/pypi/download/jsmin/2.2.2/jsmin-2.2.2.tar.gz#sha256=b6df99b2cd1c75d9d342e4335b535789b8da9107ec748212706ef7bbe5c2553b (from https://pkgs.dev.azure.com/mihaimyh/IngOne/_packaging/P10641-incoming/pypi/simple/jsmin/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
2021-09-07T11:26:05.2515374Z Collecting azure-cli
```

`Successfully installed PyJWT-1.7.1 adal-1.2.7 antlr4-python3-runtime-4.9.2 applicationinsights-0.11.10 argcomplete-1.12.3 azure-batch-8.0.0 azure-cli-2.0.73 azure-cli-command-modules-nspkg-2.0.3 azure-cli-core-2.0.73 azure-cli-nspkg-3.0.4 azure-cli-telemetry-1.0.6 azure-common-1.1.27 azure-core-1.18.0 azure-cosmos-3.2.0 azure-datalake-store-0.0.52 azure-devops-6.0.0b4 azure-functions-devops-build-0.0.22 azure-graphrbac-0.60.0 azure-identity-1.6.1 azure-keyvault-1.1.0 azure-keyvault-secrets-4.3.0 azure-mgmt-advisor-2.0.1 azure-mgmt-apimanagement-2.1.0 azure-mgmt-appconfiguration-2.0.0 azure-mgmt-applicationinsights-0.1.1 azure-mgmt-authorization-0.52.0 azure-mgmt-batch-7.0.0 azure-mgmt-batchai-2.0.0 azure-mgmt-billing-0.2.0 azure-mgmt-botservice-0.2.0 azure-mgmt-cdn-3.1.0 azure-mgmt-cognitiveservices-5.0.0 azure-mgmt-compute-6.0.0 azure-mgmt-consumption-2.0.0 azure-mgmt-containerinstance-1.5.0 azure-mgmt-containerregistry-3.0.0rc17 azure-mgmt-containerservice-5.3.0 azure-mgmt-core-1.3.0 azure-mgmt-cosmosdb-0.8.0 azure-mgmt-datalake-analytics-0.2.1 azure-mgmt-datalake-nspkg-3.0.1 azure-mgmt-datalake-store-0.5.0 azure-mgmt-datamigration-0.1.0 azure-mgmt-deploymentmanager-0.1.0 azure-mgmt-devtestlabs-2.2.0 azure-mgmt-dns-2.1.0 azure-mgmt-eventgrid-2.2.0 azure-mgmt-eventhub-2.6.0 azure-mgmt-hdinsight-1.1.0 azure-mgmt-imagebuilder-0.2.1 azure-mgmt-iotcentral-1.0.0 azure-mgmt-iothub-0.8.2 azure-mgmt-iothubprovisioningservices-0.2.0 azure-mgmt-keyvault-1.1.0 azure-mgmt-kusto-0.3.0 azure-mgmt-loganalytics-0.7.0 azure-mgmt-managedservices-1.0.0 azure-mgmt-managementgroups-0.2.0 azure-mgmt-maps-0.1.0 azure-mgmt-marketplaceordering-0.2.1 azure-mgmt-media-1.1.1 azure-mgmt-monitor-0.5.2 azure-mgmt-msi-0.2.0 azure-mgmt-netapp-0.5.0 azure-mgmt-network-4.0.0 azure-mgmt-nspkg-3.0.2 azure-mgmt-policyinsights-0.3.1 azure-mgmt-privatedns-0.1.0 azure-mgmt-rdbms-1.9.0 azure-mgmt-recoveryservices-0.4.0 azure-mgmt-recoveryservicesbackup-0.4.0 azure-mgmt-redis-6.0.0 azure-mgmt-relay-0.1.0 azure-mgmt-reservations-0.3.1 azure-mgmt-resource-3.1.0 azure-mgmt-search-2.1.0 azure-mgmt-security-0.1.0 azure-mgmt-servicebus-0.6.0 azure-mgmt-servicefabric-0.2.0 azure-mgmt-signalr-0.3.0 azure-mgmt-sql-0.12.0 azure-mgmt-sqlvirtualmachine-0.4.0 azure-mgmt-storage-4.2.0 azure-mgmt-trafficmanager-0.51.0 azure-mgmt-web-0.42.0 azure-multiapi-storage-0.2.4 azure-nspkg-3.0.2 azure-storage-blob-1.5.0 azure-storage-common-1.4.2 cryptography-2.9.2 fabric-2.6.0 humanfriendly-4.18 invoke-1.6.0 isodate-0.6.0 javaproperties-0.5.1 jsondiff-1.2.0 knack-0.6.3 mock-2.0.0 msal-1.14.0 msal-extensions-0.3.0 msrest-0.6.21 msrestazure-0.6.4 oauthlib-3.1.1 pathlib2-2.3.6 pbr-5.6.0 portalocker-1.7.1 psutil-5.8.0 pyOpenSSL-19.1.0 pydocumentdb-2.3.5 python-dateutil-2.8.2 pytz-2019.1 requests-oauthlib-1.3.0 scp-0.13.6 sshtunnel-0.1.5 tabulate-0.8.9 vsts-0.1.25 vsts-cd-manager-1.0.2 websocket-client-0.56.0 wheel-0.30.0 xmltodict-0.12.0`

### PR DESCRIPTION
**Pull request by zhoxing-ms**
Issue: #19468

**Description**

In order to solve the package conflict introduced by `jsmin`, the `jsmin` code is referenced into `json_min()`

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
ARM

File: src/azure-cli/azure/cli/command_modules/resource/_utils.py [hunk 1]
```diff
@@ -76,3 +76,222 @@ def split_resource_id(resource_id):
         return subscription_scope, relative_resource_id
 
     return "/", relative_resource_id
+
+
+def json_min(js, **kwargs):
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/resource/_utils.py:81]: I would suggest splitting `jsmin`-related functions to a separate file for easy management.

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/recordings/test_resource_group_deployment.yaml [hunk 2]
```diff
@@ -35,12 +35,12 @@ interactions:
       ParameterSetName:
       - --resource-group --template-file --parameters
       User-Agent:
-      - AZURECLI/2.26.0 azsdk-python-azure-mgmt-resource/18.0.0 Python/3.8.3 (Windows-10-10.0.18362-SP0)
+      - AZURECLI/2.28.0 azsdk-python-azure-mgmt-resource/19.0.0 Python/3.8.1 (Windows-10-10.0.19041-SP0)
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/resource/tests/latest/recordings/test_resource_group_deployment.yaml:38]: Do we really need to re-record? This code change should not change the behavior, so existing recording should work.

**Comment by jiasli** [comment]
Please include the the license of `jsmin` in `NOTICE.txt`, such as

https://github.com/Azure/azure-cli/blob/c145c29632e3ca9f1956015730aa4d698fbdd0ca/NOTICE.txt#L2663-L2686

File: src/azure-cli/azure/cli/command_modules/resource/_utils.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/_utils.py:81]: OK~

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/recordings/test_resource_group_deployment.yaml (hunk 2 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/tests/latest/recordings/test_resource_group_deployment.yaml:38]: Yes, it's not necessary, but better to have

**Comment by jiasli** [comment]
We can just release for PyPI. But be careful we shouldn't push `main` branch, otherwise `az upgrade` from other packages like `deb`, `rpm` will check the version on `main` branch (https://github.com/Azure/azure-cli/pull/19191) and think there is an update but there is actually none. This will cause endless loop in certain conditions (https://github.com/Azure/azure-cli/issues/19196).

File: src/azure-cli/azure/cli/command_modules/resource/_json_handler.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    """
+    import io
+    klass = io.StringIO
+    ins = klass(js)
+    outs = klass()
+    JsonMinify(ins, outs, **kwargs).minify()
+    return outs.getvalue()
+
+
+# pylint: disable=attribute-defined-outside-init, too-many-branches, too-many-statements, too-many-instance-attributes
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/resource/_json_handler.py:19]: We can disable all checks with

```
# pylint: skip-file
```

File: src/azure-cli/azure/cli/command_modules/resource/_json_handler.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/resource/_json_handler.py:6]: Let's put the origin of the code here: https://github.com/tikitu/jsmin/blob/master/jsmin/__init__.py

File: src/azure-cli/azure/cli/command_modules/resource/_json_handler.py (hunk 3 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/_json_handler.py:20]: I'd like to put the JSON related functions in this file in the future, so I just disable the `pylint` of this function for the time being

File: src/azure-cli/azure/cli/command_modules/resource/_json_handler.py (hunk 4 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/_json_handler.py:6]: Done

**Review by jiasli — APPROVED** [review]

**Comment by jiasli** [comment]
Kind reminder: https://github.com/tikitu/jsmin/pull/34 has been merged and [jsmin 3.0.0](https://pypi.org/project/jsmin/) has been released.

**Pull request merged**